### ISSUE TITLE
Can't stream subtitle to webos with latest version of popcorn time

### ISSUE THREAD

**Issue by bokiroki** [popcorntime/popcorntime#158]
Operating System Version: Windows 10

Popcorn Time Version: 0.3.9 build #105

Download date:

#### Expected Behaviour

...
#### Actual Behaviour

...I can't stream subtitles to lg webos tv with latest build version. When i install Build #97 (Apr 11, 2016 8:10:22 AM) version of popcorn time subtitles are working again when casting to lg webos tv.
#### Steps to repoduce the behaviour

1. ...
2. ...
3. ...

**Comment by osrl** [comment]
I can confirm this bug with two of my TV's (LG webOS, and Samsung tizen). Community edition also can stream subtitles without any problem. Console log's are not showing any error.

**Comment by liorbp** [comment]
On Popcorn Time (v0.3.9 #135) my LG WebOS is not even detected over DLNA...
This was already fixed in previous Popcorn Time project (v0.3.8) and now broken again...

**Comment by osrl** [comment]
I thought this was fixed but this bug still exist on build #166. Why is this issue closed @ChrisAlderson

**Comment by ChrisAlderson** [comment]
If the issue still persists I'll reopen the issue. Could you show the logs?

**Comment by osrl** [comment]
There is nothing wrong with the logs I think.

Series section on my web os lg:

```
[INFO] Playing next episode automatically: true
app.js:11 [INFO] Loading torrent
app.js:6 [DEBUG] Subtitles data request: Object {type: "tvshow", imdbid: "tt2085059", tvdbid: "253463", episode_id: "4201395", season: "1"…}
app.js:11 [INFO] Returning cached provider OpenSubtitles
app.js:6 [DEBUG] Streaming movie to /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf
8c64d4280f.jpg:1 GET https://walter.trakt.us/images/shows/000/041/793/fanarts/thumb/8c64d4280f.jpg 403 ()
app.js:11 [INFO] Loading torrent: startingDownload
app.js:11 [INFO] Loading torrent: downloading
app.js:11 [INFO] 25 subtitles found
app.js:6 [DEBUG] Subtitles download url: http://dl.opensubtitles.org/en/download/src-api/vrf-19c50c5b/sid-6umc450s41mkrdacne3ge77i41/filead/1953822929
app.js:6 [DEBUG] Subtitles downloaded to : /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf/Black.Mirror.S01E01.HDTV.XviD-RiVER.srt
app.js:6 [DEBUG] SUB charset detected:  windows-1252
app.js:6 [DEBUG] SUB charset expected for 'tr':  ["iso-8859-9"]
app.js:6 [DEBUG] SUB charset used:  iso-8859-9
app.js:6 [DEBUG] SubtitleServer: loading /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf/Black.Mirror.S01E01.HDTV.XviD-RiVER.srt
app.js:11 [INFO] Loading torrent: ready
app.js:11 [INFO] Loading torrent: playingExternally
app.js:11 [INFO] Device IP: 192.168.2.111
app.js:11 [INFO] Available IPs: ["fe80::1853:1fc2:5a4c:64f3","192.168.2.112","fe80::bb34:435d:4040:53d1"]
app.js:11 [INFO] 192.168.2.112 picked for external playback
app.js:11 [INFO] DLNA: play http://192.168.2.112:57780/ on '[TV][LG]42UB820V'
app.js:11 [INFO] DLNA: connecting to 192.168.2.111
2app.js:6 [DEBUG] SubtitlesServer: served vtt/srt with encoding: utf8
app.js:11 [INFO] Playing http://192.168.2.112:57780/ on [TV][LG]42UB820V
```

Series section on my Samsung tizen (Couldn't even start to stream)

```
[INFO] Playing next episode automatically: true
app.js:11 [INFO] Loading torrent
app.js:6 [DEBUG] Subtitles data request: Object {type: "tvshow", imdbid: "tt2085059", tvdbid: "253463", episode_id: "4201395", season: "1"…}
app.js:11 [INFO] Returning cached provider OpenSubtitles
app.js:6 [DEBUG] Streaming movie to /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf
8c64d4280f.jpg:1 GET https://walter.trakt.us/images/shows/000/041/793/fanarts/thumb/8c64d4280f.jpg 403 ()
app.js:11 [INFO] Loading torrent: downloading
app.js:11 [INFO] Loading torrent: waitingForSubtitles
app.js:11 [INFO] 25 subtitles found
app.js:6 [DEBUG] Subtitles download url: http://dl.opensubtitles.org/en/download/src-api/vrf-19c50c5b/sid-5q1c0icomfgni9ltrfm5k6kmf2/filead/1953822929
app.js:6 [DEBUG] Subtitles downloaded to : /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf/Black.Mirror.S01E01.HDTV.XviD-RiVER.srt
app.js:6 [DEBUG] SUB charset detected:  windows-1252
app.js:6 [DEBUG] SUB charset expected for 'tr':  ["iso-8859-9"]
app.js:6 [DEBUG] SUB charset used:  iso-8859-9
app.js:11 [INFO] Loading torrent: ready
app.js:11 [INFO] Loading torrent: playingExternally
app.js:11 [INFO] Device IP: 192.168.2.198
app.js:11 [INFO] Available IPs: ["fe80::1853:1fc2:5a4c:64f3","192.168.2.112","fe80::bb34:435d:4040:53d1"]
app.js:11 [INFO] 192.168.2.112 picked for external playback
app.js:11 [INFO] DLNA: play http://192.168.2.112:61251/ on '[TV] UE55JU7500'
app.js:11 [INFO] DLNA: connecting to 192.168.2.198
app.js:6 [DEBUG] SubtitleServer: loading /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/5e5546c30ff48e1268d4ddf635ac5577f83a87bf/Black.Mirror.S01E01.HDTV.XviD-RiVER.srt
app.js:11 [INFO] Playing http://192.168.2.112:61251/ on [TV] UE55JU7500
app.js:6 [DEBUG] SubtitlesServer: served vtt/srt with encoding: utf8
app.js:11 [INFO] Stopping external device
app.js:11 [INFO] Closing loading view
```

In movies section subtitle selector is invisible. When I select default subtitle from settings, it stucks on "Waiting for subtitle". DLNA Butter or VLC same result. There is something wrong with subtitle.

```
[INFO] Subtitles: none
app.js:11 [INFO] 40 subtitles found
app.js:11 [INFO] Returning cached provider MovieApi?&apiURL=https://movies-v2.api-fetch.website/,cloudflare+https://movies-v2.api-fetch.website,
item.js:256 Object {type: "movie", imdb_id: "tt2975590", title: "Batman v Superman: Dawn of Justice", year: 2016, genre: Array[3]…} "Movie"
app.js:11 [INFO] Show movie detail (tt2975590)
app.js:11 [INFO] Subtitles: none
app.js:11 [INFO] Loading torrent
app.js:6 [DEBUG] Subtitles data request: Object {type: "movie", imdbid: "tt2975590", filename: undefined, keywords: Array[0]}
app.js:11 [INFO] Returning cached provider OpenSubtitles
app.js:6 [DEBUG] Streaming movie to /var/folders/cj/10_7_h3j29x_ysrlbmc5jcqh0000gn/T/Butter/a0df264c995a009b422e61d3ebfab9fffbf12ad1
app.js:11 [INFO] Loading torrent: downloading
streamer.js:54 Uncaught TypeError: Cannot read property 'tr' of undefined(…)watchState @ streamer.js:54(anonymous function) @ underscore.js:768
app.js:21 [ERROR] TypeError: Cannot read property 'tr' of undefined TypeError: Cannot read property 'tr' of undefined
    at watchState (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/lib/streamer.js:54:35)
    at chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/underscore/underscore-min.js:5:8229win.error @ app.js:21(anonymous function) @ app.js:699emitOne @ events.js:96emit @ events.js:188process._fatalException @ bootstrap_node.js:313
app.js:11 [INFO] Loading torrent: waitingForSubtitles
```

**Comment by Jonnathanjons** [comment]
Hello Team This issue ( Waiting for subtitle) still exist in the latest  Popcorn Time -Desktop built #169 ( Popcorn-Time-0.3.9-win64-Setup.exe). Cannot start casting ,,shows Waiting for subtitle.Can you fix this please?

**Comment by hpinto** [comment]
i'm convinced that the subs problem is because the filename of the subtitle is not the same than the movie filename, Butter 3.10 always download the subs with the name Torrent-Downloaded-from-ExtraTorrent.cc.srt but is not what the TV (DLNA) expects. If you can orientate me where is the part of the code where butter download the subs i can try to fix that.

**Comment by Breaking1** [comment]
As you can see here i get my subtitles with the same name as the movie/episode file, i use build 239 from here https://ci.popcorntime.sh/view/All/job/Popcorn-Time-Desktop/239/

http://tinypic.com/r/2lk8uag/9

**Comment by mvuk27** [comment]
Stil not working on version 3.10 and LG WEB OS

**Comment by HamzaAlayed** [comment]
Finally, I got it work on LG webOS. this is what I did to make it works:

1. `git clone git@github.com:popcorn-official/popcorn-desktop.git`
2. `cd popcorn-desktop && gulp run`
3. `npm install`
4. `bower install`
5. `gulp build`
6. go to  popcorn-desktop/build/Popcorn-Time/osx64
7. run Popcorn-Time.app 

All these steps on Mac, I believe it will work also on linux, I am not sure how to be done on windows.
